index.bs

```diff
@@ -286,65 +290,76 @@ When the `[[track]]` of a {{MediaStreamTrackProcessor}} |processor|
+{{VideoTrackSource/track}} {{MediaStreamTrack}}.
+
+The {{VideoTrackSource}} is the [=underlying sink=] of its
+{{VideoTrackSource/writable}} attribute. The {{VideoTrackSource/track}} attribute
+is the output. Further tracks connected to the same {{VideoTrackSource}} can be
+created using the {{MediaStreamTrack/clone}} method on the
+{{VideoTrackSource/track}} attribute.
+
+The {{WritableStream}} accepts {{VideoFrame}} objects.
+When a {{VideoFrame}} is written to {{VideoTrackSource/writable}},
 the frame's `close()` method is automatically invoked, so that its internal
 resources are no longer accessible from JavaScript.
 
-### Interface definition ### {#generator-interface}
-<pre class="idl">
-[Exposed=Window,DedicatedWorker]
-interface MediaStreamTrackGenerator : MediaStreamTrack {
-    constructor(MediaStreamTrackGeneratorInit init);
-    attribute WritableStream writable;  // VideoFrame or AudioFrame
-};
+Note: There is consensus in the WG that a source capable of generating a MediaStreamTrack of kind "video" should exist.
+There is no consensus in the WG about whether or not a source capable of generating a MediaStreamTrack of kind "audio" should exist.
 
-dictionary MediaStreamTrackGeneratorInit {
-  required DOMString kind;
+
+### Interface definition ### {#video-source-interface}
+<pre class="idl">
+[Exposed=DedicatedWorker]
+interface VideoTrackSource {
+  constructor();
+  readonly attribute WritableStream writable;
+  attribute boolean muted;
+  readonly attribute MediaStreamTrack track;
 };
 </pre>
 
-### Constructor ### {#generator-constructor}
-<dfn constructor for=MediaStreamTrackGenerator title="MediaStreamTrackGenerator(init)">
-  MediaStreamTrackGenerator(init)
+Note: There is consensus in the WG that this interface should be exposed on DedicatedWorker.
+There is no consensus in the WG about whether or not it should be exposed on Window.
+
+### Internal slots ### {#internal-slots}
+<dl>
+<dt><dfn for=VideoTrackSource>`[[track]]`</dfn></dt>
+<dd>The {{MediaStreamTrack}} output of this source</dd>
+<dt><dfn for=VideoTrackSource>`[[isMuted]]`</dfn></dt>
+<dd>A boolean whose value indicates whether this source and all the
+{{MediaStreamTrack}}s it sources, are currently {{MediaStreamTrack/muted}} or not.
+</dd>
+</dl>
+
+### Constructor ### {#video-source-constructor}
+<dfn constructor for=VideoTrackSource title="VideoTrackSource(init)">
+  VideoTrackSource()
 </dfn>
-1. If |init|.{{MediaStreamTrackGeneratorInit/kind}} is not `"audio"` or `"video"`,
-    throw a {{TypeError}}.
-2. Let |g| be a new {{MediaStreamTrackGenerator}} object.
-3. Initialize the {{MediaStreamTrack/kind}} field of |g| (inherited from {{MediaStreamTrack}})
-    with |init|.{{MediaStreamTrackGeneratorInit/kind}}.
-4. Return |g|.
+1. Let |source| be a new {{VideoTrackSource}} object.
+1. Let |track| be a new {{MediaStreamTrack}} object, whose {{MediaStreamTrack/kind}} is "video", and whose {{MediaStreamTrack/id}} is a new unique id generated by the user agent.
```

Comment:
We could be using https://w3c.github.io/mediacapture-main/#dfn-initialize-the-underlying-source here to tie source and MediaStreamTrack.